pkg/providers/instance/instance.go

```diff
@@ -341,6 +385,9 @@ func (p *DefaultProvider) getOverrides(instanceTypes []*cloudprovider.InstanceTy
 	}
 	var overrides []ec2types.FleetLaunchTemplateOverridesRequest
 	for _, offering := range unwrappedOfferings {
+		if capacityReservationID != "" && offering.ReservationID() != capacityReservationID {
```

Comment:
Comment -- I believe this avoids providing overrides for reserved instance types because we have to use separate launch templates anyways?

Reply:
I'll have to verify, but I don't believe this is necessary anymore actually.